engine/src/main/java/io/zeebe/engine/processor/workflow/handlers/eventsubproc/EventSubProcessEventOccurredHandler.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.handlers.eventsubproc;
+
+import io.zeebe.engine.Loggers;
+import io.zeebe.engine.processor.workflow.BpmnStepContext;
+import io.zeebe.engine.processor.workflow.deployment.model.element.ExecutableCatchEventElement;
+import io.zeebe.engine.processor.workflow.handlers.element.EventOccurredHandler;
+import io.zeebe.engine.state.deployment.DeployedWorkflow;
+import io.zeebe.engine.state.instance.EventTrigger;
+import io.zeebe.protocol.impl.record.value.workflowinstance.WorkflowInstanceRecord;
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.protocol.record.value.BpmnElementType;
+import org.agrona.DirectBuffer;
+
+public class EventSubProcessEventOccurredHandler<T extends ExecutableCatchEventElement>
+    extends EventOccurredHandler<T> {
+  private final WorkflowInstanceRecord containerRecord = new WorkflowInstanceRecord();
+
+  public EventSubProcessEventOccurredHandler() {
+    super(null);
+  }
+
+  @Override
+  protected boolean handleState(final BpmnStepContext<T> context) {
+    final WorkflowInstanceRecord event = context.getValue();
+    final long workflowKey = event.getWorkflowKey();
+    final DeployedWorkflow workflow = context.getStateDb().getWorkflowByKey(workflowKey);
```

Comment:
We don't need to read the workflow from the state. All properties can be read from the current record.